app/celery/tasks.py

```diff
@@ -1000,3 +1011,37 @@ def get_recipient_csv(job: Job, template: Template) -> RecipientCSV:
         placeholders=template.placeholders,
         max_rows=get_csv_max_rows(job.service_id),
     )
+
+
+def _acknowledge_notification(notification_type, template, receipt):
```

Comment:
Can we please add typing to this signature?

Reply:
so I added typing to this signature, but 
```"_acknowledge_notification" has incompatible type "Union[Template, TemplateHistory, Tuple[Union[Template, TemplateHistory], Dict[Any, Any]]]"; expected "Template"```

I think this is because of the way template can be retrieved (for cache or the db)
but typing doesn't seem to be aware of the diff :/ 
I added the type check as Any :/